components/ListView/ComponentsDataListItem.js

```diff
@@ -0,0 +1,217 @@
+  DataListToggle,
+  DataListCell,
+  DataListItemCells,
+  DataListContent
+} from "@patternfly/react-core";
+import ComponentTypeIcons from "./ComponentTypeIcons";
+import ComponentSummaryList from "./ComponentSummaryList";
+
+const messages = defineMessages({
+  details: {
+    defaultMessage: "details"
+  },
+  actions: {
+    defaultMessage: "actions"
+  }
+});
+
+class ComponentsDataListItem extends React.Component {
+  constructor() {
+    super();
+    this.state = { expanded: false };
+    this.handleExpandComponent = this.handleExpandComponent.bind(this);
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (this.state.expanded === true && newProps.listItem.dependencies === undefined) {
+      this.props.fetchDetails(newProps.listItem);
+    }
+  }
+
+  handleExpandComponent() {
+    // the user clicked a list item in the blueprint contents area to expand or collapse
+    const expandState = !this.state.expanded;
+    this.setState(prevState => ({ expanded: !prevState.expanded }));
+    if (expandState === true && this.props.listItem.dependencies === undefined) {
+      this.props.fetchDetails(this.props.listItem);
+    }
+  }
+
+  render() {
+    const { listItem, noEditComponent, handleComponentDetails, handleRemoveComponent } = this.props;
+    const { expanded } = this.state;
+    const { formatMessage } = this.props.intl;
+    return (
+      <DataListItem key={listItem.name} data-component={listItem.name} isExpanded={expanded} className="cc-component">
+        <DataListItemRow>
+          <DataListToggle
+            onClick={this.handleExpandComponent}
+            isExpanded={expanded}
+            id={`${listItem.name}-toggle`}
+            aria-controls={`${listItem.name}-details`}
+            aria-label={`${listItem.name} ${formatMessage(messages.details)}`}
+          />
+          <div className="cc-c-data-list__item-icon">
+            <ComponentTypeIcons
+              componentType={listItem.ui_type}
+              componentInBlueprint={listItem.inBlueprint}
+              isSelected={listItem.userSelected}
+            />
+          </div>
+          <DataListItemCells
+            dataListCells={[
+              <DataListCell key="primary" className="cc-component__name">
+                <div>
+                  <a href="#" onClick={e => handleComponentDetails(e, listItem)}>
+                    <strong data-component-name>{listItem.name}</strong>
+                  </a>
+                </div>
+                <div>{listItem.summary}</div>
+              </DataListCell>,
+              <DataListCell
+                key="secondary"
+                className="cc-component__version pf-u-display-flex-on-xl pf-u-flex-direction-column"
+              >
+                <FormattedMessage
+                  defaultMessage="Version {version}"
+                  values={{
+                    version: <strong>{listItem.version}</strong>
+                  }}
+                  tagName={React.Fragment}
```

Comment:
This throws a prop type console error. What does the tagName prop do to the FormattedMessage?

Reply:
I don't have the link where I found this solution, but I needed the `<FormattedMessage>` to not wrap the translated string in a `<span>`. The extra `<span>` tag was interferring with the flex css applied to the contents of `<DataListCell>`. The prop `tagName` lets you specify a different html element, like `<strong>` or something else, but the suggestion was to use `React.Fragment` to essentially wrap it in `<>`, hence no tag. 

To avoid the console error, I ended up breaking the translated string (i.e. "version") from the value being passed to the string (i.e. the version value).

Reply:
Okay good to know. Thanks for the explanation.